extensions/fairifier-rdf-extension/dtls/fairifier/PostFairDataToFairDataPoint.java

```diff
@@ -36,264 +41,171 @@
+                        + catalog.getString("http://purl.org/dc/terms/hasVersion").replace(" ", "_");
+
+                catalogMetadata = getMetadata(CatalogMetadata.class, catalog,
+                        fdp.getString("baseUri") + "/catalog/" + name + "_" + uuid_catalog);
 
-                // optional
-                try {
+                if (catalog.has("http://xmlns.com/foaf/0.1/homepage")) {
                     catalogMetadata.setHomepage(f.createIRI(catalog.getString("http://xmlns.com/foaf/0.1/homepage")));
-                } catch (Exception ex) {
-                    LOGGER.warn(ex.getMessage());
                 }
+                ArrayList<IRI> catalogThemes = new ArrayList<IRI>();
                 catalogThemes.add(f.createIRI(catalog.getString("http://www.w3.org/ns/dcat#themeTaxonomy")));
                 catalogMetadata.setThemeTaxonomys(catalogThemes);
-                catalogMetadata.setTitle(f.createLiteral(catalog.getString("http://purl.org/dc/terms/title")));
-                agent.setUri(f.createIRI(fdp.getString("baseUri")));
-                agent.setName(
-                        f.createLiteral(catalog.getJSONObject("http://purl.org/dc/terms/publisher").getString("url")));
-                catalogMetadata.setPublisher(agent);
-                try {
-                    catalogMetadata.setRights(f.createIRI(catalog.getString("http://purl.org/dc/terms/rights")));
+
+                if (catalog.has("http://purl.org/dc/terms/description")) {
                     catalogMetadata
                             .setDescription(f.createLiteral(catalog.getString("http://purl.org/dc/terms/description")));
+                }
+                if (catalog.has("http://purl.org/dc/terms/language")) {
                     catalogMetadata.setLanguage(f.createIRI(catalog.getString("http://purl.org/dc/terms/language")));
+                }
+                if (catalog.has("http://xmlns.com/foaf/0.1/homepage")) {
                     catalogMetadata.setHomepage(f.createIRI(catalog.getString("http://xmlns.com/foaf/0.1/homepage")));
-                } catch (Exception ex) {
-                    LOGGER.warn(ex.getMessage());
                 }
-                catalogMetadata.setVersion(f.createLiteral(catalog.getString("http://purl.org/dc/terms/hasVersion")));
-                catalogMetadata.setUri(f.createIRI(fdp.getString("baseUri")));
-                catalogMetadata.setIssued(f.createLiteral(date));
-                catalogMetadata.setModified(f.createLiteral(date));
-                catalogString = MetadataUtils.getString(catalogMetadata, RDFFormat.TURTLE, false)
-                        .replaceAll("\\<" + catalogMetadata.getUri() + "\\>", "<>");
+                catalogPost = pushMetadataToFdp(uuid_catalog, catalogMetadata, "catalog", fdp.getString("baseUri"));
+
             }
+            DatasetMetadata datasetMetadata = null;
```

Comment:
`datasetMetadata` is not used outside of the `if` block below. This can be moved to within the block.